executor/agents/agno/agno_agent.py

```
@@ -173,10 +173,36 @@ def __init__(
+        """Return stable tool_use_id for tool_call_started events.
+
+        Uses ToolExecution.tool_call_id when available; otherwise generates a UUID
+        and caches it by object identity so the paired tool_call_completed event
+        can retrieve the same ID via _pop_tool_use_id.
+        """
+        tool_call_id = getattr(tool, "tool_call_id", "") or ""
+        if tool_call_id:
+            return tool_call_id
+        obj_id = id(tool)
+        if obj_id not in self._pending_tool_ids:
+            self._pending_tool_ids[obj_id] = str(uuid.uuid4())
```

Ruff (0.15.13)

[error] Undefined name `uuid`

(F821)